include/arch/arm/aarch64/arm_mmu.h

```diff
@@ -33,25 +33,38 @@
  * current usage of attrs[31:3] is:
  * attrs[3] : Access Permissions
  * attrs[4] : Memory access from secure/ns state
- * attrs[5] : Execute Permissions
+ * attrs[5] : Execute Permissions privileged mode (PXN)
+ * attrs[6] : Execute Permissions unprivileged mode (UXN)
+ * attrs[7] : Mirror RO/RW permissions to EL0
  *
  */
 #define MT_PERM_SHIFT		3U
 #define MT_SEC_SHIFT		4U
-#define MT_EXECUTE_SHIFT	5U
+#define MT_P_EXECUTE_SHIFT	5U
+#define MT_U_EXECUTE_SHIFT	6U
+#define MT_RW_EL0h_SHIFT	7U
 
 #define MT_RO			(0U << MT_PERM_SHIFT)
 #define MT_RW			(1U << MT_PERM_SHIFT)
 
+#define MT_RW_EL0h		(1U << MT_RW_EL0h_SHIFT)
```

Comment:
It stands for "mirror RW permissions from ELh to EL0" to be intended as "this is valid for both EL0 and ELh".

Reply:
Okay.  Its a bit of namespace collision as 'h' is always handler  ELxh-handler mode ELxt thread mode.  Though there is no EL0h still it requires little concentration to figure out its something else. You might want to give it a thought
for a more institutive name . You want to represent  AP for ELh Privileged   EL0 Unprivileged.  U_AP P_AP .. I am not sure.

Reply:
In this case `ELh` stands for `(h)igher EL`. I can use `EL0x` instead of `EL0h`.

Reply:
how about `MT_EL0_ACCESS` and `MT_CURRENT_EL_ACCESS` instead of `MT_RW_EL0h` and `MT_RW_ELh` ;
`MT_EL0_ACCESS` for EL0 access and `MT_CURRENT_EL_ACCESS` for current EL access as current EL access implied higher EL access, otherwise `MT_HIGHER_EL_ACCESS` can be used.

Reply:
The problem I see with this is that it's not possible to selectively set the AP only for EL0, so `MT_EL0_ACCESS` is not entirely correct to me. You can decide between either all the EL except EL0 or EL0 + higher EL. What about `MT_ELx_ACCESS` for EL0 + higher EL access and `MT_EL_HIGHER_ACCESS` only for higher EL?

Reply:
Pushing a new revision with:
- `MT_RW_EL0h` -> `MT_RW_AP_ELx`
- `MT_RW_ELh` -> `MT_RW_AP_EL_HIGHER`